tool/runner/runner.rs

```diff
@@ -0,0 +1,61 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::{
+    env, io,
+    path::PathBuf,
+    process::{Child, Command},
+};
+
+use tempdir::TempDir;
+
+use crate::util::{get_archive_file, unarchive};
+
+pub struct TypeDBBinaryRunner {
+    temp_dir: TempDir, // retain to prevent deletion of the directory
+    distribution: PathBuf,
+    subcommand: String,
+}
+
+impl TypeDBBinaryRunner {
```

Comment:
This runner can take the path to a TypeDB archive, which contains an entry script `<path>/typedb`, and also accepts a subcommand (eg, `console` or `server`). At `run` time, it can then be provided a list of arguments to execute with.